include/sound/sof/topology.h

```diff
@@ -147,6 +148,20 @@ struct sof_ipc_comp_src {
 	uint32_t rate_mask;	/**< SOF_RATE_ supported rates */
 } __packed;
 
+/* generic ASRC component */
+struct sof_ipc_comp_asrc {
+	struct sof_ipc_comp comp;
+	struct sof_ipc_comp_config config;
+	/* either source or sink rate must be non zero */
+	uint32_t source_rate;		/**< source rate or 0 for variable */
+	uint32_t sink_rate;		/**< sink rate or 0 for variable */
+	uint32_t asynchronous_mode;	/**< synchronous 0, asynchronous 1 */
```

Comment:
isn't this already known as asynchronous if one of the rates is zero?

Reply:
The IPC with source_rate and sink_rate is identical to SRC. We leave in topology the variable rate side parameter to value zero and set the fixed side rate. Or set both from topology if we would have e.g. a fixed 48 -> 16 kHz conversion task.

Reply:
I still don't see the benefit of the 'asynchronous mode'. It's redundant information if you already use the magic value 0 to signal the rate is variable.
And it adds error cases, e.g. if you have both rates set and async_mode as well you have an invalid configuration.

Reply:
No, I don't think it's an error to use parameters that way. We can have e.g. ASRC configured between stable 48 kHz SOF system and a drifting 16 kHz I2S to BlueTooth IC.

Also if we'd like to reduce footprint of SOF (and take a bit hit in computational load and in some cases audio quality) we could with Kconfig remove SRC and use ASRC everywhere as synchronous with drift factor locked to 1.

Reply:
Yes, it's "don't know". 0 means that it can vary, e.g. 8/16/22.05/24/32/44.1/48 and the actual rate will be retrieved from stream params. SRC component is set up similarly. Additionally ASRC in asynchronous mode tracks and adapts continuously the seen mismatch in in the Fs_out / Fs_in factor.